packages/babel-generator/src/generators/types.js

```diff
@@ -135,26 +136,11 @@ export function StringLiteral(node: Object, parent: Object) {
     return;
   }
 
-  let val = JSON.stringify(node.value);
-
-  // escape illegal js but valid json unicode characters
-  val = val.replace(/[\u000A\u000D\u2028\u2029]/g, function (c) {
-    return "\\u" + ("0000" + c.charCodeAt(0).toString(16)).slice(-4);
+  // ensure the output is ASCII-safe
+  let val = jsesc(node.value, {
+    quotes: t.isJSX(parent) ? "double" : this.format.quotes,
+    wrap: true
   });
```

Comment:
Dear god... Tracked it down further to `json_decode` returning `null` whenever there's \x in a string.